connector.go

```diff
@@ -25,6 +25,38 @@ type connector struct {
 	client *http.Client
 }
 
+type StagingCtx struct {
+	IsStagingOperation      bool
```

Comment:
What happens if the query is a GET/PUT/DELETE and the user doesn't set this to true?